core/server/fluxruntime.go

```diff
@@ -109,37 +109,63 @@ func (cs *coreServer) ListFluxCrds(ctx context.Context, msg *pb.ListFluxCrdsRequ
 		return nil, fmt.Errorf("error getting impersonating client: %w", err)
 	}
 
-	var results []*pb.Crd
+	clist := clustersmngr.NewClusteredList(func() client.ObjectList {
+		return &apiextensions.CustomResourceDefinitionList{}
+	})
 
 	respErrors := []*pb.ListError{}
 
+	if err := clustersClient.ClusteredList(ctx, clist, false); err != nil {
+		var errs clustersmngr.ClusteredListError
+
+		if !errors.As(err, &errs) {
+			return nil, fmt.Errorf("CRDs clustered list: %w", errs)
+		}
+
+		for _, e := range errs.Errors {
+			respErrors = append(respErrors, &pb.ListError{
+				ClusterName: e.Cluster,
+				Message:     e.Err.Error(),
+			})
+		}
+	}
+
+	results := []*pb.Crd{}
+
 	opts := client.MatchingLabels{
 		coretypes.PartOfLabel: FluxNamespacePartOf,
 	}
 
-	list := &apiextensions.CustomResourceDefinitionList{}
+	for clusterName, lists := range clist.Lists() {
+		for _, l := range lists {
+			list, ok := l.(*apiextensions.CustomResourceDefinitionList)
+			if !ok {
+				continue
+			}
 
-	if err := clustersClient.List(ctx, msg.ClusterName, list, opts); err != nil {
-		respErrors = append(respErrors, &pb.ListError{ClusterName: msg.ClusterName, Message: fmt.Sprintf("%s, %s", errors.New("could not list CRDs"), err)})
-	}
+			if err := clustersClient.List(ctx, clusterName, list, opts); err != nil {
+				respErrors = append(respErrors, &pb.ListError{ClusterName: clusterName, Message: fmt.Sprintf("%s, %s", errors.New("could not list CRDs"), err)})
+			}
 
-	for _, d := range list.Items {
-		version := ""
+			for _, d := range list.Items {
+				version := ""
 
-		if len(d.Spec.Versions) > 0 {
-			version = d.Spec.Versions[0].Name
-		}
+				if len(d.Spec.Versions) > 0 {
+					version = d.Spec.Versions[0].Name
+				}
 
-		r := &pb.Crd{
-			Name: &pb.Crd_Name{
-				Plural: d.Spec.Names.Plural,
-				Group:  d.Spec.Group},
-			Version:     version,
-			Kind:        d.Spec.Names.Kind,
-			ClusterName: msg.ClusterName,
-			Uid:         string(d.GetUID()),
+				r := &pb.Crd{
+					Name: &pb.Crd_Name{
+						Plural: d.Spec.Names.Plural,
+						Group:  d.Spec.Group},
```

Comment:
Formatting looks a little weird here. I would expect `gofmt` to put this end `}` on a new line? Maybe it is not that opinionated...

Reply:
`go fmt` didn't care, but yeah I agree it looks better on a new line